### PR TITLE
add repo verify command

### PR DESCRIPTION
**Pull request by whyrusleeping**
License: MIT
Signed-off-by: Jeromy <why@ipfs.io>

### CODE REVIEW

**Comment by kevina** [comment]
I would suggest that a non-zero exit code is returned from the command is some blocks failed.

Other than that it LGTM, but there are no test cases and I have not tested it manually.

File: core/commands/repo.go [hunk 1]
```diff
@@ -207,16 +212,100 @@ daemons are running.
...
+		Tagline: "Verify all blocks in repo are not corrupted.",
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		nd, err := req.InvocContext().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		out := make(chan interface{})
```
Comment (jbenet) [inline core/commands/repo.go]: buffer of 1?

Comment (jbenet) [inline core/commands/repo.go]: i guess dont need it, but usually good to do it for cases like this

File: core/commands/repo.go [hunk 2]
```diff
@@ -207,16 +212,100 @@ daemons are running.
...
+			bs.RuntimeHashing(true)
+
+			keys, err := bs.AllKeysChan(req.Context())
+			if err != nil {
+				log.Error(err)
+				return
+			}
+
+			var fails int
+			var i int
```
Comment (jbenet) [inline core/commands/repo.go]: `total` not `i`

File: test/sharness/t0084-repo-read-rehash.sh [hunk 3]
```diff
@@ -34,4 +34,9 @@ test_expect_success 'getting modified block fails' '
 	grep "block in storage has different hash than requested" err_msg
 '
 
+test_expect_success "block shows up in repo verify" '
+	test_expect_code 1 ipfs repo verify > verify_out &&
+	grep "$H_BLOCK2" verify_out
+'
```
Comment (jbenet) [inline test/sharness/t0084-repo-read-rehash.sh]: this needs much more robust testing!!
- it needs a case with hundreds of objects that all check out fine
- manipulate one object at random, verify should now fail
- fix the object, verify should now pass again

**Pull request merged**